File: Ch9/04.changeenv/changeenv.c
/*
 * changeenv.c: shows how to change the environment
 * note: calls `env` to display environments variables
*/

#include <unistd.h>

extern char ** environ;

int main() {
    char * table[3] = {NULL, NULL, NULL};  // don't forget that this string-array needs to be NULL-terminated 
    table[0] = "TERM=vt100";
    table[1] = "/on/the/range";
    
    environ = table;
    
    execlp("env", "env", NULL);
}

File: Ch9/04.changeenv/changeenv.rs
use std::env::{remove_var, set_var, vars};
use std::os::unix::process::CommandExt;
use std::process::Command;

fn main() {
    for (key, _) in vars() {
        remove_var(key);
    }

    set_var("TERM", "vt100");
    set_var("HOME", "/on/the/range");


    Command::new("env").exec();    
}
